bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java

```diff
@@ -3339,6 +3340,11 @@ private boolean checkCast(BValue sourceValue, BType targetType) {
             return checkArrayCast(sourceType, targetType);
         }
 
+        // Any to Table casting
+        if (sourceType.getTag() == TypeTags.ANY_TAG && targetType.getTag() == TypeTags.TABLE_TAG) {
+            return true;
+        }
+
```

Comment:
Do we need to add this check inside checkcast? I think this needs to be handled in ANY2DT opcode branch of the switch